src/Cli/dotnet/commands/dotnet-workload/update/WorkloadUpdateCommand.cs

```diff
@@ -139,6 +144,32 @@ public void UpdateWorkloads(bool includePreviews = false, DirectoryPath? offline
             Reporter.WriteLine();
         }
 
+        /// <summary>
+        /// Writes install records for VS Workloads so we later install the packs via the CLI for workloads managed by VS.
+        /// This is to fix a bug where updating the manifests in the CLI will cause VS to also be told to use these newer workloads via the workload resolver.
+        /// ...  but these workloads don't have their corresponding packs installed as VS doesnt update its workloads as the CLI does.
+        /// </summary>
+        private void WriteSDKInstallRecordsForVSWorkloads()
+        {
+#if !DOT_NET_BUILD_FROM_SOURCE
+            if (OperatingSystem.IsWindows())
+            {
+                // Do this gross check to avoid adding an unused & unnecessary method to FileBasedInstallers
```

Comment:
Please remove "gross" - this can trigger policheck issues.

Also, you can query whether we're an MSI or non-MSI based installation instead of looking at the actual type.

Reply:
Good point, thank you. Is that query based on the MSI sentinel file? I actually think this may be the best choice for type safety. For example, if the code to query our installation type is out of sync with the type of `_workloadInstaller` due to some bug, then this would become an invalid type cast.

Probably I prefer to do this than to have an empty function in the `FileBasedInstaller`. But if there is a better alternative, I am certainly open to it. Type checking isn't all that awful in all cases imo, I am just hesitant to use it :)